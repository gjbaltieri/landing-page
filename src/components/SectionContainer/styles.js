import Styled, { css } from 'styled-components';
export const Container = Styled.div`
${({ theme }) => css`
  max-width: 120rem;
  padding: ${theme.padding.medium};
  width: 100%;
  margin: 0 auto;
`}`;
