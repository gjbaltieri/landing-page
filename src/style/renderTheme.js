import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
