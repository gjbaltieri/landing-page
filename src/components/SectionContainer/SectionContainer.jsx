import P from 'prop-types';
import * as Styled from './styles';

export default function SectionContainer({ children }) {
  return <Styled.Container>{children}</Styled.Container>;
}

SectionContainer.prototype = {
  children: P.node.isRequired,
};
