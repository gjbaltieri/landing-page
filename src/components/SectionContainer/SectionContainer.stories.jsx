import React from 'react';
import SectionContainer from './SectionContainer';

export default {
  title: 'SectionContainer',
  component: SectionContainer,
  args: {
    children: (
      <div>
        <h1>Section Container</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem
          consectetur odit neque, qui ad at enim cum velit omnis aliquam
          corrupti quisquam ab quis voluptatum totam rerum ea eaque iste!
        </p>
      </div>
    ),
  },
  argTypes: {
    children: {
      control: {
        type: '',
      },
    },
  },
};

export const Template = (args) => <SectionContainer {...args} />;
