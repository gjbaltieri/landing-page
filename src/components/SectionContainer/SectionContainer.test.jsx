import 'jest-styled-components';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import renderTheme from '../../style/renderTheme';
import SectionContainer from './SectionContainer';

describe('<SectionContainer/>', () => {
  test('should render SectionContainer with default values', () => {
    const { container } = renderTheme(
      <SectionContainer>
        <h1>Children</h1>
      </SectionContainer>
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
