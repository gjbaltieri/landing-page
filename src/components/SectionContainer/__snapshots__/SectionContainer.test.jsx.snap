// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer/> should render SectionContainer with default values 1`] = `
.c0 {
  max-width: 120rem;
  padding: 2.4rem;
  width: 100%;
  margin: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <h1>
      hello world
    </h1>
  </div>
</div>
`;
